tests/functional/test_contracts.py

```diff
@@ -164,6 +160,54 @@ def test_contract_logs_range(contract_instance, owner, assert_log_values):
     assert_log_values(logs[0], 1)
 
 
+def test_contract_logs_range_by_address(
+    mocker, eth_tester_provider, test_accounts, contract_instance, owner, assert_log_values
+):
+    spy = mocker.spy(eth_tester_provider.web3.eth, "get_logs")
+    contract_instance.setAddress(test_accounts[1], sender=owner)
+    logs = [
+        log
+        for log in contract_instance.AddressChange.range(
+            100, event_parameters={"newAddress": test_accounts[1]}
+        )
+    ]
+
+    # NOTE: This spy assertion tests against a bug where address queries were not
```

Comment:
@NotPeopling2day OK, now the test does fail on main because this spy asserts with topics expected starting with `0x`.

This PR is GtG now!